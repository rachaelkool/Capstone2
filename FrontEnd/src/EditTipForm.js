import React, { useState, useContext, useEffect } from "react";
import { useParams, Link, useHistory } from "react-router-dom";
import UserContext from "./UserContext";
import DashboardApi from "./api/api";


function EditTipForm() {
    const { currentEmployee } = useContext(UserContext);
    const { id } = useParams();
    const [tip, setTip] = useState('');
    const history = useHistory();

    useEffect(function getCurrentTip() {
        async function getTip() {
            setTip(await DashboardApi.getTip(id));
        }
        getTip();
    }, [id]);

    const [formData, setFormData] = useState({
        content: ''
    });

    const [formErrors, setFormErrors] = useState([]);

    async function handleSubmit(e) {
        e.preventDefault();
        let employee_id = currentEmployee.empId;
        let updatedTip;
        let tipData = {
            date: tip.date,
            total_sales: formData.total_sales || tip.total_sales,
            tip_percentage: formData.tip_percentage || tip.tip_percentage,
            total_tips: formData.total_tips || tip.total_tips,
            entered_by: currentEmployee.empId
        };

        try {
            updatedTip = await DashboardApi.updateTip(id, tipData);
            alert('updated')
            history.push("/");
        } catch (e) {
            setFormErrors(e);
            return;
        }
        
        setFormData(data => ({ ...data}));
        setFormErrors([]);
        setTip(updatedTip);
    }

    function handleChange(e) {
        const { name, value } = e.target;
        setFormData(data => ({ ...data, [name]: value }));
        setFormErrors([]);
    }

    return (
        <div>
            <div>
            <form onSubmit={handleSubmit}>
                <div>
                    <label htmlFor="total_sales">Total Sales:</label>
                    <input
                        name="total_sales"
                        type="number"
                        step="0.01"
                        value={formData.total_sales}
                        placeholder={tip.total_sales}
                        onChange={handleChange}
                    />
                </div>
                <div>
                    <label htmlFor="tip_percentage">Tip Percentage:</label>
                    <input
                        name="tip_percentage"
                        type="number"
                        step="0.01"
                        value={formData.tip_percentage}
                        placeholder={tip.tip_percentage}
                        onChange={handleChange}
                    />
                </div>
                <div>
                    <label htmlFor="total_tips">Total Tips:</label>
                    <input
                        name="total_tips"
                        type="number"
                        step="0.01"
                        value={formData.total_tips}
                        placeholder={tip.total_tips}
                        onChange={handleChange}
                    />
                </div>
                <button>Update</button>
                </form>
            </div>
        </div>
    );
}


export default EditTipForm;