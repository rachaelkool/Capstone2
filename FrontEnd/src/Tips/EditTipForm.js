import React, { useState, useContext, useEffect } from "react";
import { useParams, useHistory } from "react-router-dom";
import UserContext from "../UserContext";
import DashboardApi from "../api/api";


function EditTipForm() {
    const { currentEmployee } = useContext(UserContext);
    const { id } = useParams();
    const [tip, setTip] = useState('');
    const history = useHistory();

    const [formData, setFormData] = useState({});

    useEffect(function getCurrentTip() {
        async function getTip() {
            let tip = await DashboardApi.getTip(id)
            setTip(tip);
            setFormData(tip)
        }
        getTip();
    }, [id]);

    async function handleSubmit(e) {
        e.preventDefault();
        let updatedTip;
        let tipData = {
            date: tip.date,
            total_sales: formData.total_sales || tip.total_sales,
            total_tips: formData.total_tips || tip.total_tips,
            entered_by: currentEmployee.empId
        };

        try {
            updatedTip = await DashboardApi.updateTip(id, tipData);
            alert('updated')
            history.push("/");
        } catch (e) {
            return;
        }
        
        setFormData(data => ({ ...data}));
        setTip(updatedTip);
    }

    function handleChange(e) {
        const { name, value } = e.target;
        setFormData(data => ({ ...data, [name]: value }));
    }

    return (
        <div className="feature-container">
            <h3>Input Tips</h3>
            <div className="form">
            <form className="ui form" onSubmit={handleSubmit}>
                <div className="field">
                    <label htmlFor="total_sales">Total Sales:</label>
                    <input
                        name="total_sales"
                        type="number"
                        step="0.01"
                        value={formData.total_sales}
                        onChange={handleChange}
                    />
                </div>
                <div className="field">
                    <label htmlFor="total_tips">Total Tips:</label>
                    <input
                        name="total_tips"
                        type="number"
                        step="0.01"
                        value={formData.total_tips}
                        onChange={handleChange}
                    />
                </div>
                <button>Update</button>
                </form>
            </div>
        </div>
    );
}


export default EditTipForm;