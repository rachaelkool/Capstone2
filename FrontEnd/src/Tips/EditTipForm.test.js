import React from "react";
import { render } from "@testing-library/react";
import EditTipForm from "./EditTipForm";


jest.mock('react', () => {
    const ActualReact = jest.requireActual('react')
    return {
      ...ActualReact,
      useContext: () => ({ empId: 101 }), 
    }
})
  
describe('EditTipForm', () => {
    it("renders without crashing", function() {
        render(<EditTipForm />);
    });
      
    it("matches snapshot", function() {
        const { asFragment } = render(<EditTipForm />);
        expect(asFragment()).toMatchSnapshot();
    });
})
