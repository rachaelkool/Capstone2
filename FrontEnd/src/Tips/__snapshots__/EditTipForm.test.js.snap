// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditTipForm matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="feature-container"
  >
    <h3>
      Input Tips
    </h3>
    <div
      class="form"
    >
      <form
        class="ui form"
      >
        <div
          class="field"
        >
          <label
            for="total_sales"
          >
            Total Sales:
          </label>
          <input
            name="total_sales"
            step="0.01"
            type="number"
            value=""
          />
        </div>
        <div
          class="field"
        >
          <label
            for="total_tips"
          >
            Total Tips:
          </label>
          <input
            name="total_tips"
            step="0.01"
            type="number"
            value=""
          />
        </div>
        <button>
          Update
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
